// @looker/components/dist/esm/ActionList/utils/actionListFormatting.js
import { css } from 'styled-components';
import { ActionListHeaderColumn, ActionListItemColumn, } from '..';
function filterUndefined(t) {
    return t !== undefined;
}
export const getPrimaryKeyColumnIndices = (columns) => columns
    .map((column, index) => (column.primaryKey ? index : undefined))
    .filter(filterUndefined);
export const primaryKeyColumnCSS = (columnIndices) => css `
    ${columnIndices.map((columnIndex) => css `
          ${ActionListItemColumn}:nth-child(${columnIndex + 1}) {
            color: ${(props) => props.theme.colors.palette.charcoal900};
            font-size: ${(props) => props.theme.fontSizes.small};
          }
        `)}
  `;
export const getNumericColumnIndices = (columns) => columns
    .map((column, index) => (column.type === 'number' ? index : undefined))
    .filter(filterUndefined);
export const numericColumnCSS = (columnIndices) => css `
    ${columnIndices.map((columnIndex) => css `
        ${ActionListItemColumn}:nth-child(${columnIndex + 1}),
        ${ActionListHeaderColumn}:nth-child(${columnIndex + 1}) {
          flex-direction: row-reverse
        }
      `)}
  `;
