// @looker/components/dist/cjs/utils/i18n.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const format_1 = __importDefault(require("date-fns/format"));
const isValid_1 = __importDefault(require("date-fns/isValid"));
const parse_1 = __importDefault(require("date-fns/parse"));
const ar_SA_1 = __importDefault(require("date-fns/locale/ar-SA"));
const de_1 = __importDefault(require("date-fns/locale/de"));
const en_US_1 = __importDefault(require("date-fns/locale/en-US"));
const es_1 = __importDefault(require("date-fns/locale/es"));
const fr_1 = __importDefault(require("date-fns/locale/fr"));
const it_1 = __importDefault(require("date-fns/locale/it"));
const ja_1 = __importDefault(require("date-fns/locale/ja"));
const ko_1 = __importDefault(require("date-fns/locale/ko"));
const nl_1 = __importDefault(require("date-fns/locale/nl"));
const pl_1 = __importDefault(require("date-fns/locale/pl"));
const pt_1 = __importDefault(require("date-fns/locale/pt"));
const pt_BR_1 = __importDefault(require("date-fns/locale/pt-BR"));
const ru_1 = __importDefault(require("date-fns/locale/ru"));
const sv_1 = __importDefault(require("date-fns/locale/sv"));
const tr_1 = __importDefault(require("date-fns/locale/tr"));
const zh_CN_1 = __importDefault(require("date-fns/locale/zh-CN"));
const zh_TW_1 = __importDefault(require("date-fns/locale/zh-TW"));
var Locales;
(function (Locales) {
    Locales["Arabic"] = "ar";
    Locales["German"] = "de";
    Locales["English"] = "en";
    Locales["Spanish"] = "es";
    Locales["French"] = "fr";
    Locales["Italian"] = "it";
    Locales["Japanese"] = "ja";
    Locales["Korean"] = "ko";
    Locales["Dutch"] = "nl";
    Locales["Polish"] = "pl";
    Locales["Portuguese"] = "pt";
    Locales["PortugueseBrazil"] = "pt-br";
    Locales["Russian"] = "ru";
    Locales["Swedish"] = "sv";
    Locales["Turkish"] = "tr";
    Locales["Chinese"] = "zh-cn";
    Locales["ChineseTaiwan"] = "zh-tw";
})(Locales = exports.Locales || (exports.Locales = {}));
exports.dateFnLocaleMap = {
    [Locales.Arabic]: ar_SA_1.default,
    [Locales.German]: de_1.default,
    [Locales.English]: en_US_1.default,
    [Locales.Spanish]: es_1.default,
    [Locales.French]: fr_1.default,
    [Locales.Italian]: it_1.default,
    [Locales.Japanese]: ja_1.default,
    [Locales.Korean]: ko_1.default,
    [Locales.Dutch]: nl_1.default,
    [Locales.Polish]: pl_1.default,
    [Locales.Portuguese]: pt_1.default,
    [Locales.PortugueseBrazil]: pt_BR_1.default,
    [Locales.Russian]: ru_1.default,
    [Locales.Turkish]: tr_1.default,
    [Locales.Swedish]: sv_1.default,
    [Locales.Chinese]: zh_CN_1.default,
    [Locales.ChineseTaiwan]: zh_TW_1.default,
};
exports.formatDateString = (date, locale = Locales.English) => {
    if (!date) {
        return '';
    }
    return format_1.default(date, 'P', {
        locale: exports.dateFnLocaleMap[locale],
    });
};
exports.formatYear = (date) => {
    const year = date.getFullYear();
    if (year < 100) {
        return year + 2000;
    }
    else if (year < 1000) {
        return parseInt(`${year}0`);
    }
    return year;
};
exports.parseDateFromString = (value, locale) => {
    const parsedValue = parse_1.default(value, 'P', new Date(), {
        locale: exports.dateFnLocaleMap[locale],
    });
    parsedValue.setFullYear(exports.formatYear(parsedValue));
    return isValid_1.default(parsedValue) && parsedValue;
};
