// @looker/components/dist/cjs/utils/useFocusTrap.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const focus_trap_1 = __importDefault(require("focus-trap"));
const react_1 = require("react");
const ModalContext_1 = require("../Modal/ModalContext");
const useToggle_1 = require("./useToggle");
const useCallbackRef_1 = require("./useCallbackRef");
function checkFocusLost() {
    return document.activeElement
        ? document.activeElement.tagName === 'BODY'
        : true;
}
function useFocusTrap(enabled = true, keepFocusWithin) {
    const trapRef = react_1.useRef();
    const [newElement, callbackRef] = useCallbackRef_1.useCallbackRef();
    const element = typeof keepFocusWithin === 'undefined' ? newElement : keepFocusWithin;
    const { disableFocusTrap, enableFocusTrap } = react_1.useContext(ModalContext_1.ModalContext);
    const { value, setOn, setOff } = useToggle_1.useToggle(enabled);
    react_1.useEffect(() => {
        if (element && value) {
            const autoFocusElement = element.querySelector('[data-autofocus="true"]');
            trapRef.current = focus_trap_1.default(element, Object.assign({ clickOutsideDeactivates: true, escapeDeactivates: false, fallbackFocus: element, onDeactivate: () => setOff() }, (autoFocusElement ? { initialFocus: autoFocusElement } : {})));
            disableFocusTrap && disableFocusTrap();
            window.setTimeout(() => trapRef.current && trapRef.current.activate(), 0);
        }
        else {
            trapRef.current &&
                trapRef.current.deactivate({ returnFocus: checkFocusLost() });
            enableFocusTrap && enableFocusTrap();
        }
        return () => {
            trapRef.current &&
                trapRef.current.deactivate({ returnFocus: checkFocusLost() });
        };
    }, [value, element, disableFocusTrap, enableFocusTrap, setOff]);
    return {
        callbackRef,
        disable: setOff,
        element: element || null,
        enable: setOn,
        isEnabled: value,
        trapRef,
    };
}
exports.useFocusTrap = useFocusTrap;
