// @looker/components/dist/cjs/utils/useMeasuredElement.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("react");
const resize_observer_polyfill_1 = __importDefault(require("resize-observer-polyfill"));
function measureElement(element) {
    if (!element) {
        return {
            bottom: 0,
            height: 0,
            left: 0,
            right: 0,
            top: 0,
            width: 0,
        };
    }
    return element.getBoundingClientRect();
}
exports.useMeasuredElement = (element) => {
    const [rect, setRect] = react_1.useState(measureElement());
    const handleResize = react_1.useCallback(() => {
        element && setRect(measureElement(element));
    }, [element]);
    react_1.useLayoutEffect(() => {
        if (!element) {
            return;
        }
        handleResize();
        const resizeObserver = new resize_observer_polyfill_1.default(() => handleResize());
        if (element) {
            resizeObserver.observe(element);
        }
        window.addEventListener('resize', handleResize);
        return () => {
            if (!resizeObserver) {
                return;
            }
            resizeObserver.disconnect();
            window.removeEventListener('resize', handleResize);
        };
    }, [handleResize, element]);
    return rect;
};
