// @looker/components/dist/cjs/Menu/MenuItem/MenuItem.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const merge_1 = __importDefault(require("lodash/merge"));
const react_1 = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const Icon_1 = require("../../Icon");
const MenuContext_1 = require("../MenuContext");
const MenuItemButton_1 = require("./MenuItemButton");
const MenuItemDetail_1 = require("./MenuItemDetail");
const MenuItemListItem_1 = require("./MenuItemListItem");
const menuItemStyle_1 = require("./menuItemStyle");
function useMenuItemStyleContext(props) {
    const { customizationProps: propCustomizations, compact: compactProp } = props;
    const { customizationProps: contextCustomizations, compact: contextCompact, } = react_1.useContext(MenuContext_1.MenuItemStyleContext);
    const parentCustomizations = contextCustomizations || {};
    let customizationProps = parentCustomizations || propCustomizations;
    if (customizationProps && parentCustomizations) {
        customizationProps = merge_1.default({}, parentCustomizations, propCustomizations);
    }
    const compact = compactProp === undefined ? contextCompact : compactProp;
    return { compact, customizationProps };
}
exports.useMenuItemStyleContext = useMenuItemStyleContext;
const assignCustomizations = (defaultStyle, changes) => {
    const { bg, color, iconColor, fontWeight, fontSize, iconSize } = changes || {};
    const customMarker = changes ? changes.marker : {};
    const defaults = {
        fontSize: fontSize || 'small',
        iconSize: iconSize || 20,
    };
    const base = {};
    bg && (base.bg = bg);
    color && (base.color = color);
    fontWeight && (base.fontWeight = fontWeight);
    iconColor && (base.iconColor = iconColor);
    const initial = Object.assign(Object.assign(Object.assign({}, defaultStyle.initial), defaults), base);
    const current = Object.assign(Object.assign(Object.assign(Object.assign({}, defaultStyle.current), defaults), base), (changes ? changes.current : {}));
    const hover = Object.assign(Object.assign(Object.assign(Object.assign({}, defaultStyle.hover), defaults), base), (changes ? changes.hover : {}));
    const marker = Object.assign(Object.assign({}, defaultStyle.marker), customMarker);
    return {
        current,
        hover,
        initial,
        marker,
    };
};
exports.MenuItem = (props) => {
    const { children, compact: propCompact, customizationProps: propCustomizations, current, detail, href, icon, itemRole, onBlur, onClick, onKeyUp, target } = props, remainingProps = __rest(props, ["children", "compact", "customizationProps", "current", "detail", "href", "icon", "itemRole", "onBlur", "onClick", "onKeyUp", "target"]);
    const [isFocusVisible, setFocusVisible] = react_1.useState(false);
    const handleOnKeyUp = (event) => {
        setFocusVisible(true);
        onKeyUp && onKeyUp(event);
    };
    const handleOnBlur = (event) => {
        setFocusVisible(false);
        onBlur && onBlur(event);
    };
    const { setOpen } = react_1.useContext(MenuContext_1.MenuContext);
    const handleOnClick = (event) => {
        setFocusVisible(false);
        onClick && onClick(event);
        if (setOpen && !event.defaultPrevented) {
            setOpen(false);
        }
    };
    const { customizationProps, compact } = useMenuItemStyleContext({
        compact: propCompact,
        customizationProps: propCustomizations,
    });
    const compactIconModifier = compact ? 1.25 : 1;
    const style = assignCustomizations(menuItemStyle_1.defaultMenuItemStyle, customizationProps);
    const styleState = current ? style.current : style.initial;
    const { iconSize, iconColor } = styleState, listItemProps = __rest(styleState, ["iconSize", "iconColor"]);
    const { p, py, px, pr, pl, pt, pb } = remainingProps, outerProps = __rest(remainingProps, ["p", "py", "px", "pr", "pl", "pt", "pb"]);
    const clickTargetProps = {
        p,
        pb: pb || py || p || compact ? 'xxsmall' : 'small',
        pl: pl || px || p || 'medium',
        pr: pr || px || p || 'medium',
        pt: pt || py || p || compact ? 'xxsmall' : 'small',
    };
    const { renderIconPlaceholder, setRenderIconPlaceholder } = react_1.useContext(MenuContext_1.MenuItemStyleContext);
    react_1.useEffect(() => {
        icon && setRenderIconPlaceholder(true);
    }, [icon, setRenderIconPlaceholder]);
    const renderedIcon = icon ? (react_1.default.createElement(Icon_1.Icon, { name: icon, mr: "xsmall", size: iconSize / compactIconModifier, color: iconColor })) : renderIconPlaceholder ? (react_1.default.createElement("div", null)) : undefined;
    return (react_1.default.createElement(MenuItemListItem_1.MenuItemListItem, Object.assign({ "aria-current": current && 'page', onClick: handleOnClick, current: current, focusVisible: isFocusVisible, itemStyle: style, onKeyUp: handleOnKeyUp, onBlur: handleOnBlur }, listItemProps, outerProps),
        react_1.default.createElement(MenuItemButton_1.MenuItemButton, Object.assign({ as: itemRole === 'link' ? 'a' : 'button', href: href, role: "menuitem", target: target }, clickTargetProps),
            react_1.default.createElement(MenuItemGrid, { iconSize: customizationProps.iconSize || menuItemStyle_1.defaultMenuItemStyle.initial.iconSize },
                renderedIcon,
                children)),
        detail && react_1.default.createElement(MenuItemDetail_1.MenuItemDetail, null, detail)));
};
const MenuItemGrid = styled_components_1.default.div.withConfig({ displayName: "MenuItemGrid", componentId: "sc-1sno6qj" }) `
  display: grid;
  grid-template-columns: ${({ children, iconSize }) => react_1.Children.toArray(children).length === 2 ? `${iconSize}px 1fr` : '1fr'};
  grid-gap: 0.5rem;
  align-items: center;
`;
