// @looker/components/dist/cjs/Form/Inputs/Slider/Slider.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const isFunction_1 = __importDefault(require("lodash/isFunction"));
const styled_components_1 = __importStar(require("styled-components"));
const design_tokens_1 = require("@looker/design-tokens");
const styled_system_1 = require("styled-system");
const SliderInternal = react_1.forwardRef((_a, ref) => {
    var { min = 0, max = 10, value = 0, step, onChange, name, id, className, disabled } = _a, restProps = __rest(_a, ["min", "max", "value", "step", "onChange", "name", "id", "className", "disabled"]);
    const [isFocused, setIsFocused] = react_1.useState(false);
    const [internalValue, setInternalValue] = react_1.useState(value);
    if (min > max) {
        console.warn(`Unable to render <Slider /> because the 'min' prop was set greater than 'max' value. MIN: ${min}, MAX: ${max}`);
        return null;
    }
    const boundSliderValue = (v) => Math.min(Math.max(v, min), max);
    const displayValue = isFunction_1.default(onChange)
        ? boundSliderValue(value)
        : boundSliderValue(internalValue);
    const fillPercent = Math.round(((displayValue - min) / (max - min)) * 100);
    const handleFocus = () => {
        setIsFocused(true);
    };
    const handleUnfocus = () => {
        setIsFocused(false);
    };
    const internalChangeHandler = (event) => {
        const evtValue = event.target.value;
        setInternalValue(parseInt(evtValue));
    };
    const handleChange = isFunction_1.default(onChange) ? onChange : internalChangeHandler;
    return (react_1.default.createElement("div", { className: className, "data-testid": "container" },
        react_1.default.createElement(SliderValueWrapper, null,
            react_1.default.createElement(SliderValue, { disabled: disabled, isFocused: isFocused, offsetPercent: fillPercent }, displayValue)),
        react_1.default.createElement(SliderTrack, null,
            react_1.default.createElement(SliderFill, { offsetPercent: fillPercent, disabled: disabled })),
        react_1.default.createElement(SliderInput, { disabled: disabled, id: id, isFocused: isFocused, max: max, min: min, name: name, onChange: handleChange, step: step, type: "range", value: displayValue, "aria-labelledby": restProps['aria-labelledby'], "data-testid": "slider-input", ref: ref, onBlur: handleUnfocus, onFocus: handleFocus })));
});
const sliderThumbFocusCss = styled_components_1.css `
  border-width: 5px;
`;
const sliderThumbCss = styled_components_1.css `
  border-radius: 100%;
  cursor: pointer;
  transition: transform 0.25s, box-shadow 0.25s;
  ${({ theme: { colors }, isFocused }) => styled_components_1.css `
    border: 3px solid ${colors.semanticColors.primary.main};
    height: 16px;
    width: 16px;
    background: ${colors.palette.white};
    ${isFocused && sliderThumbFocusCss}
  `}
`;
const SliderInput = styled_components_1.default.input.attrs({ type: 'range' }).withConfig({ displayName: "SliderInput", componentId: "sc-1o398ec" }) `
  background: transparent;
  display: block;
  height: 22px;
  position: relative;
  width: 100%;
  -webkit-appearance: none; /* stylelint-disable-line */

  &::-webkit-slider-thumb {
    -webkit-appearance: none; /* stylelint-disable-line */
    ${sliderThumbCss}
  }

  &::-moz-range-thumb {
    ${sliderThumbCss}
  }

  &::-ms-thumb {
    ${sliderThumbCss}
  }

  &::-ms-track {
    width: 100%;
    cursor: pointer;
    background: transparent;
    border-color: transparent;
    color: transparent;
  }

  &::-moz-focus-outer {
    border: none;
  }

  &:focus {
    outline: none;
    &::-webkit-slider-thumb {
      ${sliderThumbFocusCss}
    }

    &::-moz-range-thumb {
      ${sliderThumbFocusCss}
    }

    &::-ms-thumb {
      ${sliderThumbFocusCss}
    }
  }

  &:disabled {
    &::-webkit-slider-thumb {
      border-color: ${({ theme }) => theme.colors.palette.charcoal500};
      cursor: default;
    }
    &::-moz-range-thumb {
      border-color: ${({ theme }) => theme.colors.palette.charcoal500};
      cursor: default;
    }
    &::-ms-thumb {
      border-color: ${({ theme }) => theme.colors.palette.charcoal500};
      cursor: default;
    }
  }
`;
const SliderTrack = styled_components_1.default.div.withConfig({ displayName: "SliderTrack", componentId: "sc-a4tl27" }) `
  width: calc(100% - 16px);
  height: 4px;
  background: ${({ theme }) => theme.colors.palette.charcoal200};
  border-radius: ${({ theme }) => theme.radii.small};
  position: absolute;
  top: 50%;
  left: 8px;
  margin-top: -2px;
`;
const SliderFill = styled_components_1.default.div.withConfig({ displayName: "SliderFill", componentId: "sc-eqcw9g" }) `
  height: 100%;
  background: ${({ theme: { colors }, disabled }) => disabled ? colors.palette.charcoal400 : colors.semanticColors.primary.main};
  width: ${({ offsetPercent }) => offsetPercent}%;
  border-radius: ${({ theme }) => theme.radii.small};
`;
const SliderValue = styled_components_1.default.div.withConfig({ displayName: "SliderValue", componentId: "sc-fhyan5" }) `
  color: ${({ theme: { colors }, disabled }) => disabled ? colors.palette.charcoal700 : colors.semanticColors.primary.main};
  line-height: 1;
  user-select: none;
  transform: translateX(-50%) translateY(-1.3rem);
  left: ${({ offsetPercent }) => offsetPercent}%;
  position: absolute;
  text-align: center;
  padding: 0.2rem 0.5rem;
  border-radius: 1rem;
  background: ${({ theme, isFocused }) => isFocused ? theme.colors.palette.purple100 : theme.colors.palette.white};
`;
const SliderValueWrapper = styled_components_1.default.div.withConfig({ displayName: "SliderValueWrapper", componentId: "sc-v8qyvo" }) `
  position: relative;
  width: calc(100% - 14px);
  margin: 0 auto;
`;
exports.Slider = styled_components_1.default(SliderInternal).withConfig({ displayName: "Slider", componentId: "sc-v0u4" }) `
  ${design_tokens_1.reset}
  ${design_tokens_1.space}
  ${styled_system_1.width}
  position: relative;
`;
SliderInternal.displayName = 'Slider';
exports.Slider.defaultProps = { mt: 'large', width: '100%' };
