// @looker/components/dist/cjs/Tooltip/TooltipContent.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const styled_components_1 = __importDefault(require("styled-components"));
const Link_1 = require("../Link");
const Text_1 = require("../Text");
exports.TooltipContent = styled_components_1.default(Text_1.Paragraph).attrs((props) => ({
    fontSize: 'xsmall',
    lineHeight: 'xsmall',
    m: 'none',
    maxWidth: props.width,
    p: 'xsmall',
    width: 'auto',
})).withConfig({ displayName: "TooltipContent", componentId: "sc-18qkf37" }) `
  white-space: normal;
  text-transform: none;
  word-break: break-word;
  hyphens: auto;
  overflow-wrap: anywhere;

  ${Link_1.Link} {
    color: ${(props) => props.theme.colors.palette.blue200};
    text-decoration: underline;

    &:focus,
    &:hover {
      color: ${(props) => props.theme.colors.palette.blue100};
    }
  }
`;
exports.TooltipContent.defaultProps = { textAlign: 'center', width: '16rem' };
