// @looker/components/dist/esm/Overlay/OverlaySurfaceArrow.js
import { border, color, } from '@looker/design-tokens';
import styled from 'styled-components';
export const OverlaySurfaceArrow = styled.div.attrs((props) => ({
    'data-placement': props['data-placement'],
})).withConfig({ displayName: "OverlaySurfaceArrow", componentId: "sc-6v5kse" }) `
  position: absolute;

  &::before {
    content: '';
    display: block;
    margin: auto;
    width: 0.5rem;
    height: 0.5rem;

    ${color}
    ${border}
    /* only want border-right and border-bottom from styled-system's border */
    border-top: none;
    border-left: none;
    border-radius: 0;
  }

  &[data-placement*='top'] {
    bottom: -0.25rem;
    &::before {
      transform: rotate(45deg);
    }
  }

  &[data-placement*='right'] {
    left: -0.25rem;
    &::before {
      transform: rotate(135deg);
    }
  }

  &[data-placement*='bottom'] {
    top: -0.25rem;
    &::before {
      transform: rotate(225deg);
    }
  }

  &[data-placement*='left'] {
    right: -0.25rem;
    &::before {
      transform: rotate(315deg);
    }
  }
`;
