// @looker/components/dist/cjs/Modal/Layout/ModalContent.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const design_tokens_1 = require("@looker/design-tokens");
const react_1 = __importStar(require("react"));
const react_resize_detector_1 = __importDefault(require("react-resize-detector"));
const styled_components_1 = __importDefault(require("styled-components"));
const omit_1 = __importDefault(require("lodash/omit"));
class Internal extends react_1.Component {
    constructor(props) {
        super(props);
        this.state = { overflow: false };
        this.ref = react_1.createRef();
    }
    hasOverflow(e) {
        return e.offsetHeight < e.scrollHeight;
    }
    componentDidUpdate(prevProps) {
        if (prevProps.renderedHeight !== this.props.renderedHeight) {
            this.ref.current &&
                this.setState({ overflow: this.hasOverflow(this.ref.current) });
        }
    }
    render() {
        const _a = this.props, { children, className, innerProps } = _a, props = __rest(_a, ["children", "className", "innerProps"]);
        if (innerProps && innerProps.p && !innerProps.px) {
            innerProps.px = innerProps.p;
        }
        return (react_1.default.createElement(Outer, Object.assign({ className: `${className} ${this.state.overflow && 'overflow'}`, ref: this.ref }, omit_1.default(props, ['renderedHeight'])),
            react_1.default.createElement(Inner, Object.assign({}, innerProps), children)));
    }
}
exports.ModalContent = (props) => {
    return (react_1.default.createElement(react_resize_detector_1.default, { handleHeight: true }, (height) => react_1.default.createElement(Internal, Object.assign({ renderedHeight: height }, props))));
};
const Outer = styled_components_1.default.div.withConfig({ displayName: "Outer", componentId: "sc-azpy2a" }) `
  ${design_tokens_1.reset}
  ${design_tokens_1.layout}

  overflow: auto;
  flex: 8;

  &.overflow {
    box-shadow: inset 0 -16px 16px -16px ${design_tokens_1.palette.charcoal200},
      inset 0 16px 16px -16px ${design_tokens_1.palette.charcoal200};
  }
`;
const Inner = styled_components_1.default.div.withConfig({ displayName: "Inner", componentId: "sc-pbn4ib" }) `
  ${design_tokens_1.reset}
  ${design_tokens_1.space}
`;
Inner.defaultProps = { px: 'xlarge', py: 'large' };
