// @looker/components/dist/cjs/Text/CodeBlock.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const TextBase_1 = require("./TextBase");
const CodeBlockLayout = (_a) => {
    var { children } = _a, props = __rest(_a, ["children"]);
    return (react_1.default.createElement(TextBase_1.TextBase, Object.assign({ as: "pre" }, props),
        react_1.default.createElement("code", null, children)));
};
exports.CodeBlock = styled_components_1.default(CodeBlockLayout).withConfig({ displayName: "CodeBlock", componentId: "sc-1ksa0h5" }) `
  border: 1px solid;
  border-color: ${(props) => props.theme.colors.palette.charcoal200};
  overflow-y: scroll;
`;
exports.CodeBlock.defaultProps = {
    border: '1px solid',
    borderColor: 'palette.charcoal200',
    fontFamily: 'code',
    fontSize: 'small',
    p: 'medium',
};
