// @looker/components/dist/esm/Form/Inputs/ToggleSwitch/Knob.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { rem, rgba } from 'polished';
import React from 'react';
import styled from 'styled-components';
import { reset } from '@looker/design-tokens';
const Knob = styled(({ className }) => React.createElement("div", { className: className })).withConfig({ displayName: "Knob", componentId: "sc-1wrew84" }) `
  transform: ${({ on, size }) => (on ? `translateX(${rem(size * 0.75)})` : '')};
  transition: ${({ theme }) => theme.transitions.durationModerate};
  position: absolute;
  bottom: ${({ size }) => rem(size * 0.1)};
  left: ${({ size }) => rem(size * 0.1)};
  width: ${({ size }) => rem(size * 0.8)};
  height: ${({ size }) => rem(size * 0.8)};
  border-radius: 50%;
  background: ${({ theme }) => theme.colors.palette.white};
`;
const KnobContainerLayout = (_a) => {
    var { className } = _a, props = __rest(_a, ["className"]);
    return (React.createElement("div", { className: className },
        React.createElement(Knob, Object.assign({}, props))));
};
export const KnobContainer = styled(KnobContainerLayout).withConfig({ displayName: "KnobContainer", componentId: "sc-1cq0ej0" }) `
  ${reset}

  transition: ${({ theme }) => theme.transitions.durationModerate};
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: ${({ size }) => rem(size)};
  background: ${({ on, theme }) => on ? theme.colors.palette.purple400 : theme.colors.palette.charcoal300};

  &:hover {
    ${({ disabled, theme }) => disabled &&
    `box-shadow: 0 0 0.01rem 0.01rem ${rgba(theme.colors.palette.primary500, 0.5)}`};
  }
`;
