// @looker/components/dist/cjs/Form/Fields/FieldColor/ColorWheel/canvas_utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const d3_hsv_1 = require("d3-hsv");
const math_utils_1 = require("./math_utils");
exports.canvasMargin = 10;
exports.canvasRadius = (canvas, margin) => (canvas.width - 2 * margin) / 2;
exports.eventCartesianPosition = (canvas, event) => ({
    x: math_utils_1.translate(-canvas.getBoundingClientRect().left, event.clientX),
    y: math_utils_1.translate(-canvas.getBoundingClientRect().top, event.clientY),
});
exports.clearCanvas = (canvas) => {
    const ctx = canvas.getContext('2d');
    if (ctx) {
        ctx.clearRect(0, 0, canvas.width, canvas.height);
    }
};
exports.drawColorWheelIntoCanvasImage = (image, data) => {
    data.forEach((row, x) => {
        row.forEach((val, y) => {
            const pixelWidth = 4;
            const index = (x + y * row.length) * pixelWidth;
            const rgbColor = d3_hsv_1.hsv(val.h, val.s, val.v).rgb();
            image[index] = rgbColor.r;
            image[index + 1] = rgbColor.g;
            image[index + 2] = rgbColor.b;
            image[index + 3] = 255;
        });
    });
};
