// @looker/components/dist/cjs/ActionList/utils/actionListFormatting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const styled_components_1 = require("styled-components");
const __1 = require("..");
function filterUndefined(t) {
    return t !== undefined;
}
exports.getPrimaryKeyColumnIndices = (columns) => columns
    .map((column, index) => (column.primaryKey ? index : undefined))
    .filter(filterUndefined);
exports.primaryKeyColumnCSS = (columnIndices) => styled_components_1.css `
    ${columnIndices.map((columnIndex) => styled_components_1.css `
          ${__1.ActionListItemColumn}:nth-child(${columnIndex + 1}) {
            color: ${(props) => props.theme.colors.palette.charcoal900};
            font-size: ${(props) => props.theme.fontSizes.small};
          }
        `)}
  `;
exports.getNumericColumnIndices = (columns) => columns
    .map((column, index) => (column.type === 'number' ? index : undefined))
    .filter(filterUndefined);
exports.numericColumnCSS = (columnIndices) => styled_components_1.css `
    ${columnIndices.map((columnIndex) => styled_components_1.css `
        ${__1.ActionListItemColumn}:nth-child(${columnIndex + 1}),
        ${__1.ActionListHeaderColumn}:nth-child(${columnIndex + 1}) {
          flex-direction: row-reverse
        }
      `)}
  `;
