// @looker/components/dist/esm/utils/useScrollLock.js
import once from 'lodash/once';
import { useContext, useEffect, useRef } from 'react';
import { ModalContext } from '../Modal/ModalContext';
import { useToggle } from './useToggle';
import { useCallbackRef } from './useCallbackRef';
function setBodyOverflowHidden() {
    if (typeof document !== 'undefined') {
        document.body.style.overflow = 'hidden';
    }
}
export function useScrollLock(enabled = false, useCapture = false, allowScrollWithin) {
    const [newElement, callbackRef] = useCallbackRef();
    const element = typeof allowScrollWithin === 'undefined' ? newElement : allowScrollWithin;
    const { disableScrollLock, enableScrollLock } = useContext(ModalContext);
    const { value, setOn, setOff } = useToggle(enabled);
    const bodyOverflowRef = useRef(typeof document !== 'undefined' ? document.body.style.overflow : '');
    useEffect(() => {
        if (typeof document === 'undefined' || typeof window === 'undefined')
            return;
        let scrollTop = window.scrollY;
        let scrollTarget = document;
        const bodyOverflowCurrent = bodyOverflowRef.current;
        const setBodyOverflowOnce = once(setBodyOverflowHidden);
        function stopScroll(e) {
            setBodyOverflowOnce();
            if (e.target !== null && e.target !== scrollTarget) {
                scrollTarget = e.target;
                scrollTop =
                    scrollTarget instanceof Element
                        ? scrollTarget.scrollTop
                        : window.scrollY;
            }
            if (scrollTarget instanceof Element &&
                !(element && element.contains(scrollTarget))) {
                scrollTarget.scrollTop = scrollTop;
            }
            else if (scrollTarget === document) {
                window.scrollTo({ top: scrollTop });
            }
        }
        if (element && value) {
            window.addEventListener('scroll', stopScroll, true);
            disableScrollLock && disableScrollLock();
            setBodyOverflowHidden();
        }
        else {
            window.removeEventListener('scroll', stopScroll, true);
            enableScrollLock && enableScrollLock();
            document.body.style.overflow = bodyOverflowCurrent;
        }
        return () => {
            window.removeEventListener('scroll', stopScroll, true);
            document.body.style.overflow = bodyOverflowCurrent;
        };
    }, [value, element, useCapture, disableScrollLock, enableScrollLock]);
    return {
        callbackRef,
        disable: setOff,
        element: element || null,
        enable: setOn,
        isEnabled: value,
    };
}
