// @looker/components/dist/cjs/Overlay/OverlaySurfaceArrow.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const design_tokens_1 = require("@looker/design-tokens");
const styled_components_1 = __importDefault(require("styled-components"));
exports.OverlaySurfaceArrow = styled_components_1.default.div.attrs((props) => ({
    'data-placement': props['data-placement'],
})).withConfig({ displayName: "OverlaySurfaceArrow", componentId: "sc-6v5kse" }) `
  position: absolute;

  &::before {
    content: '';
    display: block;
    margin: auto;
    width: 0.5rem;
    height: 0.5rem;

    ${design_tokens_1.color}
    ${design_tokens_1.border}
    /* only want border-right and border-bottom from styled-system's border */
    border-top: none;
    border-left: none;
    border-radius: 0;
  }

  &[data-placement*='top'] {
    bottom: -0.25rem;
    &::before {
      transform: rotate(45deg);
    }
  }

  &[data-placement*='right'] {
    left: -0.25rem;
    &::before {
      transform: rotate(135deg);
    }
  }

  &[data-placement*='bottom'] {
    top: -0.25rem;
    &::before {
      transform: rotate(225deg);
    }
  }

  &[data-placement*='left'] {
    right: -0.25rem;
    &::before {
      transform: rotate(315deg);
    }
  }
`;
