// @looker/components/dist/esm/Text/CodeBlock.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from 'react';
import styled from 'styled-components';
import { TextBase } from './TextBase';
const CodeBlockLayout = (_a) => {
    var { children } = _a, props = __rest(_a, ["children"]);
    return (React.createElement(TextBase, Object.assign({ as: "pre" }, props),
        React.createElement("code", null, children)));
};
export const CodeBlock = styled(CodeBlockLayout).withConfig({ displayName: "CodeBlock", componentId: "sc-1ksa0h5" }) `
  border: 1px solid;
  border-color: ${(props) => props.theme.colors.palette.charcoal200};
  overflow-y: scroll;
`;
CodeBlock.defaultProps = {
    border: '1px solid',
    borderColor: 'palette.charcoal200',
    fontFamily: 'code',
    fontSize: 'small',
    p: 'medium',
};
