// @looker/components/dist/cjs/Calendar/calendar-size.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const styled_system_1 = require("styled-system");
exports.daySizeMap = {
    small: 32,
    medium: 36,
    large: 44,
};
exports.calendarSize = styled_system_1.variant({
    prop: 'size',
    variants: {
        small: {
            fontSize: 'xsmall',
            height: `${exports.daySizeMap.small}px`,
            width: `${exports.daySizeMap.small}px`,
        },
        medium: {
            fontSize: 'small',
            height: `${exports.daySizeMap.medium}px`,
            width: `${exports.daySizeMap.medium}px`,
        },
        large: {
            fontSize: 'medium',
            height: `${exports.daySizeMap.large}px`,
            width: `${exports.daySizeMap.large}px`,
        },
    },
});
exports.calendarSpacing = styled_system_1.variant({
    prop: 'size',
    variants: {
        small: {
            p: 'small',
        },
        medium: {
            p: 'medium',
        },
        large: {
            p: 'large',
        },
    },
});
