// @looker/components/dist/esm/Avatar/AvatarCombo.js
import React from 'react';
import styled from 'styled-components';
import { Icon } from '../Icon';
import { AvatarUser } from './AvatarUser';
import { AvatarIcon } from './AvatarIcon';
const AvatarIconSecondary = styled(AvatarIcon).withConfig({ displayName: "AvatarIconSecondary", componentId: "sc-uvvioe" }) ``;
const AvatarLayout = ({ secondaryIcon, secondaryColor, secondaryBg, color, icon = 'User', user, className, }) => {
    return (React.createElement("div", { className: className },
        user ? (React.createElement(AvatarUser, { user: user, color: color })) : (React.createElement(AvatarIcon, { color: color, icon: icon })),
        React.createElement(AvatarIconSecondary, { bg: secondaryBg, color: secondaryColor, icon: secondaryIcon })));
};
export const AvatarCombo = styled(AvatarLayout).withConfig({ displayName: "AvatarCombo", componentId: "sc-15cgr3l" }) `
  position: relative;
  height: 40px;
  width: 40px;

  ${AvatarIconSecondary} {
    bottom: -4px;
    right: -4px;
    position: absolute;
    height: 20px;
    width: 20px;

    ${Icon} {
      height: 14px;
      width: 14px;
    }
  }
`;
