// @looker/components/dist/esm/Avatar/Avatar.js
import { color, reset, space, typography, } from '@looker/design-tokens';
import { css } from 'styled-components';
import { variant } from 'styled-system';
const size = variant({
    prop: 'size',
    variants: {
        xxsmall: {
            fontSize: 'xsmall',
            height: '24px',
            width: '24px',
        },
        xsmall: {
            fontSize: 'xsmall',
            height: '32px',
            width: '32px',
        },
        small: {
            fontSize: 'small',
            height: '40px',
            width: '40px',
        },
        medium: {
            fontSize: 'medium',
            height: '54px',
            width: '54px',
        },
        large: {
            fontSize: 'large',
            height: '60px',
            width: '60px',
        },
    },
});
export const avatarCSS = css `
  ${reset}

  ${color}
  ${space}
  ${typography}
  ${size}

  align-items: center;
  border-radius: 100%;
  display: grid;
  justify-items: center;
  overflow: hidden;
`;
