// @looker/sdk/lib/rtl/authToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthToken = void 0;
class AuthToken {
    constructor(token) {
        this.access_token = '';
        this.token_type = '';
        this.expires_in = 0;
        this.expiresAt = new Date();
        if (!token)
            token = {};
        this.setToken(token);
    }
    isActive() {
        if (!this.access_token)
            return false;
        if (!this.expiresAt)
            return false;
        const now = new Date();
        return this.expiresAt > now;
    }
    setToken(token) {
        this.access_token = token.access_token || '';
        this.token_type = token.token_type || '';
        this.expires_in = token.expires_in || 0;
        if (token.refresh_token) {
            this.refresh_token = token.refresh_token;
        }
        const exp = new Date();
        if (this.access_token && this.expires_in) {
            exp.setSeconds(exp.getSeconds() + this.expires_in);
        }
        else {
            exp.setSeconds(exp.getSeconds() - 10);
        }
        this.expiresAt = exp;
        return this;
    }
    reset() {
        this.access_token = '';
        this.expires_in = 0;
    }
}
exports.AuthToken = AuthToken;
