// @looker/sdk/lib/rtl/browserSession.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BrowserSession = void 0;
const browserTransport_1 = require("./browserTransport");
const authSession_1 = require("./authSession");
class BrowserSession extends authSession_1.AuthSession {
    constructor(settings, transport) {
        super(settings, transport || new browserTransport_1.BrowserTransport(settings));
        this.settings = settings;
        this._activeToken = '';
    }
    get activeToken() {
        if (!this._activeToken) {
            const meta = document.head.querySelector('[name=csrf-token]');
            this._activeToken = meta ? meta.content : '';
        }
        return this._activeToken;
    }
    async getToken() {
        return this.activeToken;
    }
    isAuthenticated() {
        const token = this.activeToken;
        if (!token)
            return false;
        return true;
    }
    async authenticate(props) {
        const token = this.activeToken;
        if (token)
            props.headers['X-CSRF-TOKEN'] = token;
        return props;
    }
}
exports.BrowserSession = BrowserSession;
