// @looker/sdk/lib/rtl/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unquote = exports.boolDefault = exports.isFalse = exports.isTrue = exports.matchModeBinary = exports.matchModeString = exports.matchCharsetUtf8 = exports.matchCharset = exports.environmentPrefix = exports.sdkVersion = exports.apiVersion = exports.lookerVersion = exports.defaultApiVersion = void 0;
exports.defaultApiVersion = '3.1';
exports.lookerVersion = '7.6';
exports.apiVersion = '4.0';
exports.sdkVersion = `${exports.apiVersion}.${exports.lookerVersion}`;
exports.environmentPrefix = 'LOOKERSDK';
exports.matchCharset = ';.*charset=';
exports.matchCharsetUtf8 = `${exports.matchCharset}.*\\butf-8\\b`;
exports.matchModeString = `(^application\\/.*(\\bjson\\b|\\bxml\\b|\\bsql\\b|\\bgraphql\\b|\\bjavascript\\b|\\bx-www-form-urlencoded\\b)|^text\\/|.*\\+xml\\b|${exports.matchCharset})`;
exports.matchModeBinary = '^image\\/|^audio\\/|^video\\/|^font\\/|^application\\/|^multipart\\/';
exports.isTrue = (value) => /^(true|t|yes|y|1)$/i.test(value);
exports.isFalse = (value) => /^(false|f|no|n|0)$/i.test(value);
exports.boolDefault = (value, defaultBool = false) => {
    if (exports.isTrue(value))
        return true;
    if (exports.isFalse(value))
        return false;
    return defaultBool;
};
exports.unquote = (value) => {
    if (!value)
        return '';
    if (/^['\"`]/.test(value)) {
        const quote = value.substring(0, 1);
        if (value.endsWith(quote))
            return value.substring(1, value.length - 1);
    }
    return value;
};
