// @looker/sdk/lib/rtl/corsSession.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CorsSession = void 0;
const authSession_1 = require("./authSession");
const browserTransport_1 = require("./browserTransport");
const transport_1 = require("./transport");
const authToken_1 = require("./authToken");
class CorsSession extends authSession_1.AuthSession {
    constructor(settings, transport) {
        super(settings, transport || new browserTransport_1.BrowserTransport(settings));
        this.settings = settings;
        this.activeToken = new authToken_1.AuthToken();
    }
    isAuthenticated() {
        return this.activeToken.isActive();
    }
    async authenticate(props) {
        if (!this.isAuthenticated()) {
            const token = await this.getToken();
            if (token) {
                this.activeToken.setToken(token);
            }
        }
        if (this.isAuthenticated()) {
            props.mode = 'cors';
            delete props.credentials;
            props.headers = {
                Authorization: `Bearer ${this.activeToken.access_token}`,
                [transport_1.LookerAppId]: transport_1.agentPrefix,
            };
        }
        return props;
    }
}
exports.CorsSession = CorsSession;
