// @looker/sdk/lib/rtl/extensionSession.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionSession = void 0;
class ExtensionSession {
    constructor(settings, transport) {
        this.settings = settings;
        this.sudoId = '';
        this.settings = settings;
        this.transport = transport;
    }
    isAuthenticated() {
        return true;
    }
    async authenticate(init) {
        return Promise.reject(Error(`Authenticate ${init ? 'request property overrides' : ''} not supported from ExtensionSession`));
    }
    async getToken() {
        return Promise.reject(Error('Access to token is not allowed from ExtensionSession'));
    }
    isSudo() {
        throw new Error('isSudo is not allowed from ExtensionSession');
    }
    async login(sudoId) {
        return Promise.reject(Error(`Login ${sudoId ? 'for sudo' : ''} not supported from ExtensionSession`));
    }
    async logout() {
        return Promise.reject(Error('Logout not supported from ExtensionSession'));
    }
    reset() {
    }
}
exports.ExtensionSession = ExtensionSession;
