// @looker/sdk/lib/rtl/extensionTransport.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionTransport = void 0;
class ExtensionTransport {
    constructor(options, hostConnection) {
        this.options = options;
        this.hostConnection = hostConnection;
        this.options = options;
        this.hostConnection = hostConnection;
    }
    async rawRequest(method, path, queryParams, body, authenticator, options) {
        return this.hostConnection.rawRequest(method, path, body, queryParams, authenticator, options);
    }
    async request(method, path, queryParams, body, authenticator, options) {
        return this.hostConnection.request(method, path, body, queryParams, authenticator, options);
    }
    async stream(callback, method, path, queryParams, body, authenticator, options) {
        options = options ? Object.assign(Object.assign({}, this.options), options) : this.options;
        return this.hostConnection.stream(callback, method, path, body, queryParams, authenticator, options);
    }
}
exports.ExtensionTransport = ExtensionTransport;
