// @looker/components/dist/cjs/Form/Inputs/Radio/FauxRadio.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const styled_components_1 = __importDefault(require("styled-components"));
const design_tokens_1 = require("@looker/design-tokens");
const dotSize = 6;
exports.FauxRadio = styled_components_1.default.div.withConfig({ displayName: "FauxRadio", componentId: "sc-18jidrp" }) `
  ${design_tokens_1.reset}
  position: relative;
  width: 100%;
  height: 100%;
  background-color: currentColor;
  border: solid 1px ${({ theme }) => `${theme.colors.palette.charcoal200}`};
  border-radius: 50%;
  border-color: currentColor;
  color: transparent;
  transition: background-color 25ms linear, border-color 25ms linear,
    box-shadow 25ms linear;

  &::after {
    content: '';
    position: absolute;
    background: #fff;
    width: ${dotSize}px;
    height: ${dotSize}px;
    border-radius: 50%;
    top: calc(50% - ${dotSize / 2}px);
    right: calc(50% - ${dotSize / 2}px);
  }
`;
