// @looker/components/dist/cjs/Spinner/SpinnerMarker.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const design_tokens_1 = require("@looker/design-tokens");
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
const markerStyle = (props) => {
    const { markerIndex, markerRadius, markers, speed } = props;
    const delay = (markerIndex * speed) / markers;
    const rotateAngle = (360 / markers) * markerIndex;
    return styled_components_1.css `
    animation: ${design_tokens_1.quarterFade} ${speed}ms linear ${delay}ms infinite;
    border-radius: ${markerRadius && `${markerRadius}px`};
    transform: rotate(${rotateAngle}deg) translate(0, -160%);
  `;
};
exports.SpinnerMarker = styled_components_1.default.div.withConfig({ displayName: "SpinnerMarker", componentId: "sc-13zccv2" }) `
  ${styled_system_1.color}
  ${markerStyle}
  height: 20%;
  left: 48%;
  opacity: 0.25;
  position: absolute;
  top: 40%;
  width: 6%;
`;
