// @looker/components/dist/cjs/Sidebar/SidebarGroup.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const Text_1 = require("../Text");
const Layout_1 = require("../Layout");
const Icon_1 = require("../Icon");
const InternalSidebarGroup = (_a) => {
    var { children, className, label, showChildren = false } = _a, props = __rest(_a, ["children", "className", "label", "showChildren"]);
    const [isOpen, setOpen] = react_1.useState(showChildren);
    const toggle = () => {
        setOpen(!isOpen);
        return false;
    };
    return (react_1.default.createElement("section", Object.assign({ className: className }, props),
        react_1.default.createElement(SidebarGroupHeading, { onClick: toggle },
            react_1.default.createElement("button", { "aria-expanded": isOpen ? 'true' : 'false' },
                label,
                react_1.default.createElement(Icon_1.Icon, { size: 20, name: isOpen ? 'CaretUp' : 'CaretDown' }))),
        isOpen && react_1.default.createElement(Layout_1.Box, { px: "medium" }, children)));
};
const SidebarGroupHeading = styled_components_1.default(Text_1.Heading).attrs({
    as: 'h3',
    fontSize: 'small',
    fontWeight: 'semiBold',
}).withConfig({ displayName: "SidebarGroupHeading", componentId: "sc-1nqspgx" }) `
  button {
    align-items: center;
    all: inherit;
    display: flex;
    justify-content: space-between;
    width: 100%;
    background: transparent;
    border: none;

    ${Icon_1.Icon} {
      align-self: center;
    }
  }
`;
exports.SidebarGroup = styled_components_1.default(InternalSidebarGroup).withConfig({ displayName: "SidebarGroup", componentId: "sc-et365l" }) ``;
