// @looker/components/dist/cjs/Avatar/Avatar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const design_tokens_1 = require("@looker/design-tokens");
const styled_components_1 = require("styled-components");
const styled_system_1 = require("styled-system");
const size = styled_system_1.variant({
    prop: 'size',
    variants: {
        xxsmall: {
            fontSize: 'xsmall',
            height: '24px',
            width: '24px',
        },
        xsmall: {
            fontSize: 'xsmall',
            height: '32px',
            width: '32px',
        },
        small: {
            fontSize: 'small',
            height: '40px',
            width: '40px',
        },
        medium: {
            fontSize: 'medium',
            height: '54px',
            width: '54px',
        },
        large: {
            fontSize: 'large',
            height: '60px',
            width: '60px',
        },
    },
});
exports.avatarCSS = styled_components_1.css `
  ${design_tokens_1.reset}

  ${design_tokens_1.color}
  ${design_tokens_1.space}
  ${design_tokens_1.typography}
  ${size}

  align-items: center;
  border-radius: 100%;
  display: grid;
  justify-items: center;
  overflow: hidden;
`;
