// @looker/components/dist/cjs/Modal/ModalBackdrop.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const design_tokens_1 = require("@looker/design-tokens");
const styled_system_1 = require("styled-system");
const styled_components_1 = __importDefault(require("styled-components"));
exports.ModalBackdrop = styled_components_1.default.div.attrs((props) => ({
    backgroundColor: props.visible ? props.backgroundColor : 'transparent',
})).withConfig({ displayName: "ModalBackdrop", componentId: "sc-b0r9rx" }) `
  ${design_tokens_1.reset}

  ${styled_system_1.color}
  cursor: default;
  opacity: ${(props) => props.opacity || 0.4};
  transition: opacity ${(props) => props.theme.transitions.durationSimple};
  ${(props) => props.inlineStyle};

  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  position: fixed;

  &.entering,
  &.exiting {
    opacity: 0.01;
  }
`;
exports.ModalBackdrop.defaultProps = {
    backgroundColor: 'palette.charcoal200',
    opacity: 0.6,
    visible: true,
};
