// @looker/components/dist/cjs/Modal/ModalPortal.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const react_dom_1 = require("react-dom");
const styled_components_1 = __importDefault(require("styled-components"));
const modalRoot_1 = require("./modalRoot");
exports.CustomizableModalAttributes = {};
exports.ModalPortal = react_1.forwardRef(({ children }, ref) => {
    const el = react_1.useRef(document.createElement('div'));
    react_1.useEffect(() => {
        const modalRoot = modalRoot_1.getModalRoot();
        if (!modalRoot)
            return;
        const elCurrent = el.current;
        modalRoot.appendChild(elCurrent);
        return () => {
            modalRoot.removeChild(elCurrent);
        };
    }, [el]);
    const content = (react_1.default.createElement(InvisiBox, { ref: ref, tabIndex: -1, zIndex: exports.CustomizableModalAttributes.zIndex }, children));
    return react_dom_1.createPortal(content, el.current);
});
exports.ModalPortal.displayName = 'ModalPortal';
const InvisiBox = styled_components_1.default.div.withConfig({ displayName: "InvisiBox", componentId: "sc-16z53yg" }) `
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;

  align-items: center;
  display: flex;
  justify-content: center;
  pointer-events: none;
  z-index: ${(props) => props.zIndex};

  * {
    pointer-events: auto;
  }
`;
